velox/expression/tests/RowWriterTest.cpp

```diff
@@ -0,0 +1,390 @@
+    auto& rowWriter = vectorWriter.current();
+    rowWriter.set_null_at<0>();
+    vectorWriter.commit();
+  }
+
+  auto expected =
+      vectorMaker_.flatVectorNullable<StringView>({{"hi"_sv, std::nullopt}});
+  assertEqualVectors(result, makeRowVector({expected}));
+}
+
+TEST_F(RowWriterTest, varBinary) {
+  auto result = prepareResult(makeRowType({VARBINARY()}), 2);
+
+  exec::VectorWriter<RowWriterT<Varchar>> vectorWriter;
+  vectorWriter.init(*result->as<RowVector>());
+
+  {
+    vectorWriter.setOffset(0);
+    auto& rowWriter = vectorWriter.current();
+    rowWriter.get_writer_at<0>().append("hi");
+    vectorWriter.commit();
+  }
+
+  {
+    vectorWriter.setOffset(1);
+    auto& rowWriter = vectorWriter.current();
+    rowWriter.set_null_at<0>();
+    vectorWriter.commit();
+  }
+
+  // Test results.
+  DecodedVector decoded;
+  SelectivityVector rows(result->size());
+  decoded.decode(*result, rows);
+  exec::VectorReader<Row<Varbinary>> reader(&decoded);
+  ASSERT_EQ(exec::get<0>(reader[0]), "hi"_sv);
+  ASSERT_EQ(exec::get<0>(reader[1]), std::nullopt);
+}
+
+TEST_F(RowWriterTest, nested) {
+  // Output is row(row(int, string), double).
+  auto result = prepareResult(
+      makeRowType({makeRowType({BIGINT(), VARCHAR()}), DOUBLE()}));
+
+  exec::VectorWriter<RowWriterT<RowWriterT<int64_t, Varchar>, double>>
+      vectorWriter;
+  vectorWriter.init(*result->as<RowVector>());
+
+  vectorWriter.setOffset(0);
+  auto& outerRowWriter = vectorWriter.current();
+
+  auto& innerRowWriter = outerRowWriter.get_writer_at<0>();
+  innerRowWriter.get_writer_at<0>() = 1;
+  innerRowWriter.get_writer_at<1>().append("hi");
+
+  outerRowWriter.get_writer_at<1>() = 1.1;
+  vectorWriter.commit();
+
+  auto expected1 = vectorMaker_.flatVectorNullable<int64_t>({{1}});
+  auto expected2 = vectorMaker_.flatVectorNullable<StringView>({{"hi"_sv}});
+  auto expected3 = vectorMaker_.flatVectorNullable<double>({{1.1}});
+  assertEqualVectors(
+      result,
+      makeRowVector({makeRowVector({expected1, expected2}), expected3}));
+}
+
+TEST_F(RowWriterTest, rowOfArray) {
+  auto result = prepareResult(makeRowType({ARRAY(BIGINT())}));
+
+  exec::VectorWriter<RowWriterT<ArrayWriterT<int64_t>>> vectorWriter;
+  vectorWriter.init(*result->as<RowVector>());
+
+  vectorWriter.setOffset(0);
+
+  auto& rowWriter = vectorWriter.current();
+  auto& arrayWriter = rowWriter.get_writer_at<0>();
+  arrayWriter.push_back(1);
+  arrayWriter.push_back(2);
+  arrayWriter.push_back(3);
+
+  vectorWriter.commit();
+
+  auto expected = vectorMaker_.arrayVector<int64_t>({{1, 2, 3}});
+  assertEqualVectors(result, makeRowVector({expected}));
+}
+
+TEST_F(RowWriterTest, arrayOfRows) {
+  auto result = prepareResult(ARRAY(makeRowType({BIGINT(), BIGINT()})));
+
+  exec::VectorWriter<ArrayWriterT<RowWriterT<int64_t, int64_t>>> vectorWriter;
+  vectorWriter.init(*result->as<ArrayVector>());
+
+  vectorWriter.setOffset(0);
+
+  auto& arratWriter = vectorWriter.current();
```

Comment:
nit: arratWriter -> arrayWriter (I think this was copied into other tests in this file too)